src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala

```diff
@@ -108,27 +108,29 @@ abstract class AgoraService(permissionsDataSource: PermissionsDataSource) extend
   // GET http://root.com/configurations/<namespace>/<name>/<snapshotId>
   def querySingleRoute =
     matchQuerySingleRoute(path) { (namespace, name, snapshotId, username) =>
-      extractOnlyPayloadParameter { (onlyPayload) =>
-        val targetEntity = AgoraEntity(Option(namespace), Option(name), Option(snapshotId))
-
-        get { requestContext =>
-          completeWithPerRequest(requestContext, targetEntity, username, toBool(onlyPayload), path, queryHandlerProps)
-        } ~
-        delete { requestContext =>
-          completeEntityDelete(requestContext, targetEntity, username, path, queryHandlerProps)
-        } ~
-        post {
-          // only allow copying (post) for methods
-          if (path != AgoraConfig.methodsRoute) {
-            reject(MethodRejection(HttpMethods.GET), MethodRejection(HttpMethods.DELETE))
-          } else {
-            entity(as[AgoraEntity]) { newEntity =>
-              AgoraEntity.validate(newEntity) match {
-                case Success(_) => //noop
-                case Failure(errors) => throw new IllegalArgumentException(s"Method is invalid: Errors: $errors")
-              }
-              parameters("redact".as[Boolean] ? false) { redact =>
-                requestContext => completeEntityCopy(requestContext, targetEntity, newEntity, redact, username, path, queryHandlerProps)
+      extractOnlyPayloadParameter { (onlyPayload: Option[String]) =>
+        extractPayloadAsObjectParameter { (payloadAsObject: Option[String]) =>
```

Comment:
In the yaml, it's defined as a boolean. Is it intentional to unmarshal it as a string here?

Reply:
The Swagger type just defines what kind of control gets rendered for that parameter. For optional booleans, it's a dropdown limited to the empty string, `true`, and `false`.

The DSL syntax you suggested is awesome, gladly adopting!